### ISSUE TITLE
"-y" is not a valid package manager. Available package managers are: npm, cnpm, pnpm, yarn or bun.

### ISSUE THREAD

**Issue by tomerh2001** [oven-sh/bun#6075]
### What version of Bun is running?

1.0.3

### What platform is your computer?

Linux 5.15.90.1-microsoft-standard-WSL2 x86_64 x86_64

### What steps can reproduce the bug?

1. run `bun add only-allow`
2. add the following to your `package.json` scripts:
```
    "preinstall": "npx -y only-allow bun",
```
3. run `bun install`:
4. 
```
root@laptop:/app/mccan-workspace/projects/routines# bun install
bun install v1.0.3 (25e69c71)
"-y" is not a valid package manager. Available package managers are: npm, cnpm, pnpm, yarn or bun.
```

### What is the expected behavior?

To be able to use bun with `only-allow`

### What do you see instead?

"-y" is not a valid package manager. Available package managers are: npm, cnpm, pnpm, yarn or bun.

### Additional information

After adding a `only-allow` preinstall, bun fails

**Comment by nektro** [comment]
is this still impacting you? it appears to be working on my end in Bun 1.1.10 (you may need to run `bun upgrade`)

**Comment by dylan-conway** [comment]
This is fixed as of bun `v1.0.31`
